pkg/controllers/cassandra/pilot/pilot_test.go

```diff
@@ -32,114 +35,120 @@ func nonClusterPod(cluster *v1alpha1.CassandraCluster, name string) *v1.Pod {
+				if pilotCount != expectedPilotCount {
+					t.Log(pilots.Items)
+					t.Errorf("Unexpected pilot count: %d != %d", expectedPilotCount, pilotCount)
+				}
+			},
 		},
-	)
-	t.Run(
-		"pilot sync when hash changes",
-		func(t *testing.T) {
-			f := casstesting.NewFixture(t)
-			pod := clusterPod(f.Cluster, "foo")
-			unsyncedPilot := pilot.PilotForCluster(f.Cluster, pod)
-			pilot.UpdateHashAnnotation(unsyncedPilot, 0)
-			f.AddObjectK(pod)
-			f.AddObjectN(unsyncedPilot)
-			f.Run()
-			f.AssertPilotsLength(1)
-			updatedPilot := f.Pilots().Items[0]
-			updatedPilotAnnotations := updatedPilot.GetAnnotations()
-			hash, ok := updatedPilotAnnotations[pilot.HashAnnotationKey]
-			if !ok {
-				t.Log(updatedPilotAnnotations)
-				t.Error("pilot hash annotation not found")
-			}
-			if hash == "0" {
-				t.Log(updatedPilot)
-				t.Error("Pilot was not updated")
-			}
+		"non-cluster pods are ignored": {
+			kubeObjects: []runtime.Object{
+				clusterPod(cluster1, "pod1"),
+				nonClusterPod(cluster1, "pod3"),
+			},
+			cluster: cluster1,
+			assertions: func(t *testing.T, state *controllers.State) {
+				pilots, err := state.NavigatorClientset.
+					Navigator().Pilots(cluster1.Namespace).List(metav1.ListOptions{})
+				if err != nil {
+					t.Fatal(err)
+				}
+				expectedPilotCount := 1
+				pilotCount := len(pilots.Items)
+				if pilotCount != expectedPilotCount {
+					t.Log(pilots.Items)
+					t.Errorf("Unexpected pilot count: %d != %d", expectedPilotCount, pilotCount)
+				}
+			},
 		},
-	)
-	t.Run(
-		"pilot no sync if hash matches",
-		func(t *testing.T) {
-			f := casstesting.NewFixture(t)
-			pod := clusterPod(f.Cluster, "foo")
-			// Remove the labels
-			unsyncedPilot := pilot.PilotForCluster(f.Cluster, pod)
-			unsyncedPilot.SetLabels(map[string]string{})
-			pilot.ComputeHashAndUpdateAnnotation(unsyncedPilot)
-			f.AddObjectK(pod)
-			f.AddObjectN(unsyncedPilot)
-			f.Run()
-			f.AssertPilotsLength(1)
-			updatedPilot := f.Pilots().Items[0]
-			updatedLabels := updatedPilot.GetLabels()
-			if len(updatedLabels) == 0 {
-				t.Log(updatedPilot)
-				t.Error("pilot was not updated")
-			}
+		"no error if pilot exists": {
+			kubeObjects: []runtime.Object{
+				clusterPod(cluster1, "pod1"),
+			},
+			navObjects: []runtime.Object{
+				pilot.PilotForCluster(cluster1, clusterPod(cluster1, "pod1")),
+			},
+			cluster: cluster1,
 		},
-	)
-	t.Run(
-		"don't clobber custom labels",
-		func(t *testing.T) {
-			f := casstesting.NewFixture(t)
-			pod := clusterPod(f.Cluster, "foo")
-			// Remove the labels
-			unsyncedPilot := pilot.PilotForCluster(f.Cluster, pod)
-			unsyncedPilot.Labels["foo"] = "bar"
-			f.AddObjectK(pod)
-			f.AddObjectN(unsyncedPilot)
-			f.Run()
-			f.AssertPilotsLength(1)
-			updatedPilot := f.Pilots().Items[0]
-			updatedLabels := updatedPilot.GetLabels()
-			if updatedLabels["foo"] != "bar" {
-				t.Log(updatedLabels)
-				t.Error("custom labels were altered")
-			}
+		"no error if pilot not yet listed": {
```

Comment:
This *should* be an error IMO (we have retry logic to handle cache inconsistencies)